lib/segment/src/vector_storage/multi_dense/simple_multi_dense_vector_storage.rs

```diff
@@ -134,19 +134,19 @@ fn open_simple_multi_dense_vector_storage_impl<T: PrimitiveVectorElement>(
         }
 
         let metadata = &mut vectors_metadata[point_id_usize];
-        metadata.size = stored_record.vector.flattened_vectors.len();
-        metadata.capacity = metadata.size;
+        metadata.inner_vectors_count = stored_record.vector.vector_count();
```

Comment:
the bug was at this line, assigning the flatten len as a size for the vector count